lib/jormungandr/test/bench/Latency.hs

```diff
@@ -341,6 +431,46 @@ extractTimings isStart isFinish msgs = map2 mkDiff filtered
         _ -> Nothing
     getTimestamp = tstamp . loMeta
 
+measureObserveLogs
+    :: TVar [LogObject ServerLog] -- ^ Log message variable.
+    -> IO a -- ^ Action to run
+    -> IO [NominalDiffTime]
+measureObserveLogs = measureDBLatency
+
+-- | Measure how long an action takes based on trace points of bracketObserveIO
+-- and taking an average of results over a short time period.
+measureDBLatency
+    :: Show msg => TVar [LogObject msg] -- ^ Log message variable.
+    -> IO a -- ^ Action to run
+    -> IO [NominalDiffTime]
+measureDBLatency tvar action = do
+    atomically $ writeTVar tvar []
+    _res <- repeatFor sampleTimeSeconds action
+    extractDBTimings . reverse <$> readTVarIO tvar
+
+-- | Scan through iohk-monitoring logs and extract time differences between
+-- ObserveOpen and ObserveClose emitted while using bracketObserveIO.
+extractDBTimings
+    :: Show a => [LogObject a] -- ^ Log messages
+    -> [NominalDiffTime]
+extractDBTimings msgs = map2 mkDiff filtered
+  where
+    map2 _ [] = []
+    map2 f (a:b:xs) = (f a b:map2 f xs)
+    map2 _ _ = error "start trace without matching finish trace"
+
+    mkDiff (False, start) (True, finish) = diffUTCTime finish start
+    mkDiff (False, _) _ = error "missing finish trace"
+    mkDiff (True, _) _ = error "missing start trace"
+
+    filtered = mapMaybe filterMsg msgs
+    filterMsg logObj = case loContent logObj of
+        ObserveOpen _ -> Just (False, getTimestamp logObj)
+        ObserveClose _ -> Just (True, getTimestamp logObj)
```

Comment:
tried to use the fact that `bracketObserveIO` emits `ObserveOpen` and `ObserveClose` and we use it inside runQuery in Sqlite.hs. Unfortunately, I do not see such logs at all (I listed and searched through all). Although, I see for example, 
```
LogObject {loName = ["database","query"], loMeta = LOMeta {tstamp = 2019-12-12 14:49:22.995996905 UTC, tid = "18", hostname = "", severity = Debug, privacy = Public}, loContent = LogMessage (LogText "SELECT \"wallet_id\", \"slot\", \"input_tx_id\", \"input_index\", \"output_address\", \"output_coin\" FROM \"utxo\" WHERE (\"wallet_id\"=?) AND (\"slot\"=?)")}
```
and of course ApiLogs like : 
```
LogObject {loName = ["api"], loMeta = LOMeta {tstamp = 2019-12-12 14:49:23.006473579 UTC, tid = "18", hostname = "", severity = Debug, privacy = Public}, loContent = LogMessage (LogApiServerMsg (ApiLog {requestId = RequestId 525, logMsg = LogRequestFinish}))}
```
@rvl I you have an idea how to rely on `ObserveOpen/Close` then please share, otherwise I am thinking about extending DBLog and add `QueryStart`, QueryFinish` and do as it is done for `ApiLog`